gtsam/hybrid/IncrementalHybrid.cpp

```diff
@@ -32,23 +33,30 @@ void gtsam::IncrementalHybrid::update(gtsam::GaussianHybridFactorGraph graph,
     for (auto &&conditional : *hybridBayesNet_) {
       for (auto &key : conditional->frontals()) {
         if (allVars.find(key) != allVars.end()) {
-          if (auto
-              gf = boost::dynamic_pointer_cast<GaussianMixture>(conditional)) {
+          if (auto gf =
+                  boost::dynamic_pointer_cast<GaussianMixture>(conditional)) {
             graph.push_back(gf);
-          } else if (auto df =
-              boost::dynamic_pointer_cast<DiscreteConditional>(conditional)) {
+          } else if (auto df = boost::dynamic_pointer_cast<DiscreteConditional>(
+                         conditional)) {
             graph.push_back(df);
           }
           break;
         }
       }
     }
+  } else {
+    // Initialize an empty HybridBayesNet
+    hybridBayesNet_ = boost::make_shared<HybridBayesNet>();
```

Comment:
Why a shared pointer? In another PR I am trying to eradicate all shared pointers to what are essentially vectors. Actually, not looking for a reason, but a change :-)

Reply:
PS, line 29 would have to be changed as well -> just check if empty